engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -249,23 +236,36 @@ private void preRenderUpdate(DefaultRenderingProcess.StereoRenderState stereoRen
             renderableWorld.update();
             renderableWorld.generateVBOs();
             secondsSinceLastFrame = 0;
+
+            updateMaterials(); // TODO: this should happen only if necessary, i.e. upon notification from the Asset system.
         }
 
-        if (stereoRenderState != DefaultRenderingProcess.StereoRenderState.MONO) {
+        if (currentRenderingStage != WorldRenderingStage.MONO) {
             playerCamera.updateFrustum();
         }
 
-        // this line needs to be here as deep down it relies on the camera's frustrum.
+        // this line needs to be here as deep down it relies on the camera's frustrum, updated just above.
         renderableWorld.queueVisibleChunks(isFirstRenderingStageForCurrentFrame);
     }
 
     /**
+     * This allows for materials to be swapped at runtime, i.e. if a module provides new ones.
+     */
+    private void updateMaterials() {
+        // TODO: review - perhaps only the chunk shader would ever need to be swapped at runtime? All the others could be final?
+        chunkShader         = Assets.getMaterial("engine:prog.chunk");
+        lightGeometryShader = Assets.getMaterial("engine:prog.lightGeometryPass");
+        //simpleShader        = Assets.getMaterial("engine:prog.simple");  // in use by the currently commented out light stencil pass
+        shadowMapShader     = Assets.getMaterial("engine:prog.shadowMap");
+    }
```

Comment:
These Material/shader assignments were previously deeper in the rendering methods. Moving them here doesn't really make much of a difference for the time being: if the shaders are swapped at runtime, theoretically the renderer will make immediate use of them, but fetching from Assets happens nevertheless once per frame.  

However, if an Assets-related entity can emit an event when one or more Shader change, it'd be easy to pack this method within a conditional, so that it gets executed only upon receiving said event. Is there such facility already perhaps?